### PR TITLE
Update note card and other CSS

### LINKED ISSUE #217: Beautify country notes card
Currently the country notes card is quite ugly. Hence, this issue was created.
Might need to add additional notes-related attributes such as Tags. 

Tagging @rtshkmr since he is also implementing some parts of Note. (no pun intended)

### LINKED ISSUE #250: Fix client card display issue
Probably just need to add more padding to the clientlist

image

### PR DESCRIPTION
**Pull request by qwoprocks**
## Description

Fixes #217, fixes #250 

Changes (view ss below for reference): 
* Added proper note card (supports tags) to country note list panel
* Changed the menuitems on the top bar so that the text can be seen (previously was white text on grey/white bg)
* Refactored country note list panel to use a ScollPane, and a VBox inside it containing all the notes that updates using a observablelist listener
* Made it so that Tags can only be max 45 characters long

image

## Testing

Ran all test, passed all tests.
Ran the following commands:
`country note delete 2`
`country note add c/US nt/blahblahabalmecor ncomn`
`country note view c/US`
`country note delete 1`
`client edit 1 t/pneumonoultramicroscopicsilicovolcanoconiosis` (success)
`client edit 1 t/pneumonoultramicroscopicsilicovolcanoconiosism` (fail)
`client add n/mmm p/901 e/m@c.com a/bac,lem c/SG tz/GMT+9 t/pneumonoultramicroscopicsilicovolcanoconiosis t/pneumonoultramicroscopicsilicovolcanoconiosimmm` (fail)
`client add n/mmm p/901 e/m@c.com a/bac,lem c/SG tz/GMT+9 t/pneumonoultramicroscopicsilicovolcanoconiosis t/pneumonoultramicroscopicsilicovolcanoconios` (success)

## Remarks

Not sure if we want to use the same way for clients as I did for notes (refer to point 3 of changes). @tankangliang @raysonkoh @rtshkmr @LeeEnHao what do y'all think? The current listview has buggy behavior with scrolling, if you have one cell that's much longer than the others, the scrollbar gets smaller when that cell is rendered, and then bigger again as you scroll on. Its most obvious when you click and drag the scrollbar.

### CODE REVIEW

**Review by raysonkoh — APPROVED** [review]
Manually tested GUI, LGTM.

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 1]
```diff
@@ -9,7 +9,9 @@
  */
 public class Tag {
 
-    public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
+    public static final int MAX_CHARACTERS = 45;
+    public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric and be within " + MAX_CHARACTERS
```
Comment (tankangliang) [inline src/main/java/seedu/address/model/tag/Tag.java]: Maybe rephrase to "maximum of 45 characters" instead of "be within 45 characters"?

File: src/main/java/seedu/address/ui/CountryNoteListPanel.java [hunk 2]
```diff
@@ -1,36 +1,92 @@
...
+    private static final String HEADER_ALL_COUNTRIES_TEXT = "All Country Notes";
+
+    private boolean displayAllCountries = true;
+    private ObservableList<CountryNote> countryNoteObservableList;
+
     @FXML
     private Label header;
     @FXML
-    private ListView<CountryNote> countryNoteListView;
+    private VBox countryNoteListView;
```
Comment (tankangliang) [inline src/main/java/seedu/address/ui/CountryNoteListPanel.java:28]: Should this be `countryListView` since the subpanel contains a `countryNoteListView`?

**Review by tankangliang — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/ui/CountryNoteListPanel.java (hunk 2 above)
Comment (qwoprocks) [reply src/main/java/seedu/address/ui/CountryNoteListPanel.java:28]: This one is being used to display country notes if we're looking at one country only, so I just left it as countryNoteListView

File: src/main/java/seedu/address/model/tag/Tag.java (hunk 1 above)
Comment (qwoprocks) [reply src/main/java/seedu/address/model/tag/Tag.java]: done

**Review by LeeEnHao — APPROVED** [review]
Looks good. There's still a slight aesthetic issue where the list cell is not contained within the list view rounded container, but this is pretty trivial.

image

**Comment by qwoprocks** [comment]
> 
> 
> Looks good. There's still a slight aesthetic issue where the list cell is not contained within the list view rounded container, but this is pretty trivial.
> 
> image

good catch, I'll add it as an issue

**Review by tankangliang — APPROVED** [review]
